### ISSUE TITLE
Linux - Contributor Support Thread

### ISSUE THREAD

**Issue by tin2tin** [tin2tin/Pallaidium#105]
As I do not personally use Linux, I can't offer support for it, but if anyone has problems with it, report the Linux-related bugs here, and let's hope some other Linux users can help out, submitting fixes for those problems.

**Comment by vuaieo** [comment]
hi i tryed to install the pallaidium addon from latest main branch and i get this error when trying to install dependencies:

Ensuring: pip
Traceback (most recent call last):
  File "/home/vuaieo/.config/upbge/4.3/scripts/addons/Pallaidium-main/__init__.py", line 990, in execute
    install_modules(self)
  File "/home/vuaieo/.config/upbge/4.3/scripts/addons/Pallaidium-main/__init__.py", line 708, in install_modules
    subprocess.call([pybin, "-m", "ensurepip"])
  File "/home/vuaieo/Desktop/UPBGE_29_8_2024/4.3/python/lib/python3.11/subprocess.py", line 389, in call
    with Popen(*popenargs, **kwargs) as p:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vuaieo/Desktop/UPBGE_29_8_2024/4.3/python/lib/python3.11/subprocess.py", line 1026, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/home/vuaieo/Desktop/UPBGE_29_8_2024/4.3/python/lib/python3.11/subprocess.py", line 1950, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: '/home/vuaieo/Desktop/UPBGE_29_8_2024/4.3/python/bin/python'
Error: Python: Traceback (most recent call last):
  File "/home/vuaieo/.config/upbge/4.3/scripts/addons/Pallaidium-main/__init__.py", line 708, in install_modules
    subprocess.call([pybin, "-m", "ensurepip"])
  File "/home/vuaieo/Desktop/UPBGE_29_8_2024/4.3/python/lib/python3.11/subprocess.py", line 389, in call
    with Popen(*popenargs, **kwargs) as p:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vuaieo/Desktop/UPBGE_29_8_2024/4.3/python/lib/python3.11/subprocess.py", line 1026, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/home/vuaieo/Desktop/UPBGE_29_8_2024/4.3/python/lib/python3.11/subprocess.py", line 1950, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: '/home/vuaieo/Desktop/UPBGE_29_8_2024/4.3/python/bin/python'

**Comment by tin2tin** [comment]
Try a vanilla version of Blender instead of UPBGE (I have no idea what changes are made in this fork of Blender).

**Comment by vuaieo** [comment]
hi ok i installed official blender and addon again and i get same messege i think:

Ensuring: pip
Traceback (most recent call last):
  File "/home/vuaieo/.config/blender/4.2/scripts/addons/Pallaidium-main/__init__.py", line 990, in execute
    install_modules(self)
  File "/home/vuaieo/.config/blender/4.2/scripts/addons/Pallaidium-main/__init__.py", line 708, in install_modules
    subprocess.call([pybin, "-m", "ensurepip"])
  File "/home/vuaieo/Desktop/blender-4.2.1-linux-x64/4.2/python/lib/python3.11/subprocess.py", line 389, in call
    with Popen(*popenargs, **kwargs) as p:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vuaieo/Desktop/blender-4.2.1-linux-x64/4.2/python/lib/python3.11/subprocess.py", line 1026, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/home/vuaieo/Desktop/blender-4.2.1-linux-x64/4.2/python/lib/python3.11/subprocess.py", line 1950, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: '/home/vuaieo/Desktop/blender-4.2.1-linux-x64/4.2/python/bin/python'
Error: Python: Traceback (most recent call last):
  File "/home/vuaieo/.config/blender/4.2/scripts/addons/Pallaidium-main/__init__.py", line 708, in install_modules
    subprocess.call([pybin, "-m", "ensurepip"])
  File "/home/vuaieo/Desktop/blender-4.2.1-linux-x64/4.2/python/lib/python3.11/subprocess.py", line 389, in call
    with Popen(*popenargs, **kwargs) as p:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vuaieo/Desktop/blender-4.2.1-linux-x64/4.2/python/lib/python3.11/subprocess.py", line 1026, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/home/vuaieo/Desktop/blender-4.2.1-linux-x64/4.2/python/lib/python3.11/subprocess.py", line 1950, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: '/home/vuaieo/Desktop/blender-4.2.1-linux-x64/4.2/python/bin/python'

**Comment by tin2tin** [comment]
As you know, I'm not on Linux. Do you feel up for getting your hands dirty? (I can tell you how to disable the subprocess.call([pybin, "-m", "ensurepip"]) line - which is nice to have - but not nessessary)

**Comment by vuaieo** [comment]
absolutely !!! how?

**Comment by tin2tin** [comment]
In Blender Preferences, check this and save Preferences in lover left corner - burger menu:
![image](https://github.com/user-attachments/assets/09d55f15-5a9d-4fd0-92a5-128c3d8d89d4)

In the tabs, click the + and add a scripting workspace:
![image](https://github.com/user-attachments/assets/3f2964ed-d779-4d5a-96df-7257a224e8da)

Right click on the Pallaidium generate button, and select Edit Source:
![image](https://github.com/user-attachments/assets/f183b6ee-462c-4250-9a26-7c2ea3d569f9)

In the Text Editor select `__init__.py `:
![image](https://github.com/user-attachments/assets/86902403-6f40-4012-96db-f6d7b098b96a)

This is the (messy) source code of Pallaidium.

**Comment by tin2tin** [comment]
Now highlight the ensure pip related lines and comment them (so they'll become disabled):
![image](https://github.com/user-attachments/assets/95af0482-23bc-4a17-8dcb-af2a81f5defa)

Then save the script: Alt+S (keep the mouse cursor in the Text Editor area when saving with shortcut)

Hit F3 and type reload scripts (not strips!!!):
![image](https://github.com/user-attachments/assets/0f6700d2-1435-44ec-885a-1e14a52ef61a)

When you hit Return, Pallaidium will be updated with this change you made. And hopefully the error will be gone.

**Comment by tin2tin** [comment]
Does this location have a file called python? /home/vuaieo/Desktop/UPBGE_29_8_2024/4.3/python/bin/
If not, where is it?

**Comment by vuaieo** [comment]
yes there was but this python file is called "python3.11"

**Comment by jurassicjordan** [comment]
I followed the workaround above and successfully (or at least I didnt get any errors after/while installing deps) installed the addon. I went ahead to test just stable diffusion XL and got the following error

`Python: Traceback (most recent call last):
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/diffusers/utils/import_utils.py", line 830, in _get_module
    return importlib.import_module("." + module_name, self.__name__)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1126, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1147, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/diffusers/models/autoencoders/__init__.py", line 1, in <module>
    from .autoencoder_asym_kl import AsymmetricAutoencoderKL
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/diffusers/models/autoencoders/autoencoder_asym_kl.py", line 21, in <module>
    from ..modeling_outputs import AutoencoderKLOutput
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/diffusers/models/modeling_outputs.py", line 7, in <module>
    class AutoencoderKLOutput(BaseOutput):
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/diffusers/utils/outputs.py", line 70, in __init_subclass__
    torch.utils._pytree.register_pytree_node(
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: module 'torch.utils._pytree' has no attribute 'register_pytree_node'. Did you mean: '_register_pytree_node'?

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/diffusers/utils/import_utils.py", line 830, in _get_module
    return importlib.import_module("." + module_name, self.__name__)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1147, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/diffusers/pipelines/pipeline_utils.py", line 44, in <module>
    from ..models import AutoencoderKL
  File "<frozen importlib._bootstrap>", line 1229, in _handle_fromlist
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/diffusers/utils/import_utils.py", line 820, in __getattr__
    module = self._get_module(self._class_to_module[name])
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/diffusers/utils/import_utils.py", line 832, in _get_module
    raise RuntimeError(
RuntimeError: Failed to import diffusers.models.autoencoders.autoencoder_kl because of the following error (look up to see its traceback):
module 'torch.utils._pytree' has no attribute 'register_pytree_node'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/jordancruz/.config/blender/4.2/scripts/addons/Pallaidium-main/__init__.py", line 3679, in execute
    from diffusers import DiffusionPipeline, DPMSolverMultistepScheduler
  File "<frozen importlib._bootstrap>", line 1229, in _handle_fromlist
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/diffusers/utils/import_utils.py", line 821, in __getattr__
    value = getattr(module, name)
            ^^^^^^^^^^^^^^^^^^^^^
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/diffusers/utils/import_utils.py", line 820, in __getattr__
    module = self._get_module(self._class_to_module[name])
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/diffusers/utils/import_utils.py", line 832, in _get_module
    raise RuntimeError(
RuntimeError: Failed to import diffusers.pipelines.pipeline_utils because of the following error (look up to see its traceback):
Failed to import diffusers.models.autoencoders.autoencoder_kl because of the following error (look up to see its traceback):
module 'torch.utils._pytree' has no attribute 'register_pytree_node'
`

**Comment by tin2tin** [comment]
@jurassicjordan Did you remember to restart after installing the dependencies? What is your Linux distro?

Googling this part: `'torch.utils._pytree' has no attribute 'register_pytree_node` seems to indicate the transformers python lib does not have the right modules. And you could either need a newer version or an older version. 

Currently, is the Pallaidium transformers version specified, because it works on Windows(and maybe on some Linux versions?): 
`transformers==4.43.0`

What you could do is to install this add-on: https://github.com/amb/blender_pip

Then type in "transformers" and click uninstall. And then click Install and not if it is a newer version being installed. Then restart, and check if it is working now. If not, you can try earlier versions, by writing like this syntax: transformers==4.43.0 and hit Install. 

Here you can see the version numbers: https://pypi.org/project/transformers/#history

If you find a version number working for you, let me know (I do not run Linux myself).

**Comment by jurassicjordan** [comment]
I did restart after dependencies where installed. I also tried to install that addon, but Blender complained that there was no manifest. I’ll try installing it manually by unzipping into the addons folder and see if I can install that version of transformers

**Comment by jurassicjordan** [comment]
okay, managed to get the pip addon installed, and successfully installed transformers  4.43.0 but I'm still getting a...Slightly? differnent but same error I think?? see at the end of the error it says something about an undefined symbol. didn't say that last time:
```
Python: Traceback (most recent call last):
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/transformers/utils/import_utils.py", line 1586, in _get_module
    return importlib.import_module("." + module_name, self.__name__)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1147, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/transformers/models/clip/modeling_clip.py", line 44, in <module>
    from ...modeling_flash_attention_utils import _flash_attention_forward
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/transformers/modeling_flash_attention_utils.py", line 27, in <module>
    from flash_attn.bert_padding import index_first_axis, pad_input, unpad_input  # noqa
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/flash_attn/__init__.py", line 3, in <module>
    from flash_attn.flash_attn_interface import (
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/flash_attn/flash_attn_interface.py", line 10, in <module>
    import flash_attn_2_cuda as flash_attn_cuda
ImportError: /home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/flash_attn_2_cuda.cpython-311-x86_64-linux-gnu.so: undefined symbol: _ZN2at4_ops5zeros4callEN3c108ArrayRefINS2_6SymIntEEENS2_8optionalINS2_10ScalarTypeEEENS6_INS2_6LayoutEEENS6_INS2_6DeviceEEENS6_IbEE

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/diffusers/utils/import_utils.py", line 830, in _get_module
    return importlib.import_module("." + module_name, self.__name__)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1147, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/diffusers/loaders/unet.py", line 46, in <module>
    from .lora_pipeline import LORA_WEIGHT_NAME, LORA_WEIGHT_NAME_SAFE, TEXT_ENCODER_NAME, UNET_NAME
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/diffusers/loaders/lora_pipeline.py", line 43, in <module>
    from ..models.lora import text_encoder_attn_modules, text_encoder_mlp_modules
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/diffusers/models/lora.py", line 35, in <module>
    from transformers import CLIPTextModel, CLIPTextModelWithProjection
  File "<frozen importlib._bootstrap>", line 1229, in _handle_fromlist
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/transformers/utils/import_utils.py", line 1577, in __getattr__
    value = getattr(module, name)
            ^^^^^^^^^^^^^^^^^^^^^
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/transformers/utils/import_utils.py", line 1576, in __getattr__
    module = self._get_module(self._class_to_module[name])
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/transformers/utils/import_utils.py", line 1588, in _get_module
    raise RuntimeError(
RuntimeError: Failed to import transformers.models.clip.modeling_clip because of the following error (look up to see its traceback):
/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/flash_attn_2_cuda.cpython-311-x86_64-linux-gnu.so: undefined symbol: _ZN2at4_ops5zeros4callEN3c108ArrayRefINS2_6SymIntEEENS2_8optionalINS2_10ScalarTypeEEENS6_INS2_6LayoutEEENS6_INS2_6DeviceEEENS6_IbEE

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/diffusers/utils/import_utils.py", line 830, in _get_module
    return importlib.import_module("." + module_name, self.__name__)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1126, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1147, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/diffusers/models/autoencoders/__init__.py", line 1, in <module>
    from .autoencoder_asym_kl import AsymmetricAutoencoderKL
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/diffusers/models/autoencoders/autoencoder_asym_kl.py", line 23, in <module>
    from .vae import DecoderOutput, DiagonalGaussianDistribution, Encoder, MaskConditionDecoder
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/diffusers/models/autoencoders/vae.py", line 25, in <module>
    from ..unets.unet_2d_blocks import (
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/diffusers/models/unets/__init__.py", line 7, in <module>
    from .unet_2d_condition import UNet2DConditionModel
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/diffusers/models/unets/unet_2d_condition.py", line 22, in <module>
    from ...loaders import PeftAdapterMixin, UNet2DConditionLoadersMixin
  File "<frozen importlib._bootstrap>", line 1229, in _handle_fromlist
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/diffusers/utils/import_utils.py", line 820, in __getattr__
    module = self._get_module(self._class_to_module[name])
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/diffusers/utils/import_utils.py", line 832, in _get_module
    raise RuntimeError(
RuntimeError: Failed to import diffusers.loaders.unet because of the following error (look up to see its traceback):
Failed to import transformers.models.clip.modeling_clip because of the following error (look up to see its traceback):
/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/flash_attn_2_cuda.cpython-311-x86_64-linux-gnu.so: undefined symbol: _ZN2at4_ops5zeros4callEN3c108ArrayRefINS2_6SymIntEEENS2_8optionalINS2_10ScalarTypeEEENS6_INS2_6LayoutEEENS6_INS2_6DeviceEEENS6_IbEE

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/diffusers/utils/import_utils.py", line 830, in _get_module
    return importlib.import_module("." + module_name, self.__name__)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1147, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/diffusers/pipelines/pipeline_utils.py", line 44, in <module>
    from ..models import AutoencoderKL
  File "<frozen importlib._bootstrap>", line 1229, in _handle_fromlist
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/diffusers/utils/import_utils.py", line 820, in __getattr__
    module = self._get_module(self._class_to_module[name])
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/diffusers/utils/import_utils.py", line 832, in _get_module
    raise RuntimeError(
RuntimeError: Failed to import diffusers.models.autoencoders.autoencoder_kl because of the following error (look up to see its traceback):
Failed to import diffusers.loaders.unet because of the following error (look up to see its traceback):
Failed to import transformers.models.clip.modeling_clip because of the following error (look up to see its traceback):
/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/flash_attn_2_cuda.cpython-311-x86_64-linux-gnu.so: undefined symbol: _ZN2at4_ops5zeros4callEN3c108ArrayRefINS2_6SymIntEEENS2_8optionalINS2_10ScalarTypeEEENS6_INS2_6LayoutEEENS6_INS2_6DeviceEEENS6_IbEE

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/jordancruz/.config/blender/4.2/scripts/addons/Pallaidium-main/__init__.py", line 3679, in execute
    from diffusers import DiffusionPipeline, DPMSolverMultistepScheduler
  File "<frozen importlib._bootstrap>", line 1229, in _handle_fromlist
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/diffusers/utils/import_utils.py", line 821, in __getattr__
    value = getattr(module, name)
            ^^^^^^^^^^^^^^^^^^^^^
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/diffusers/utils/import_utils.py", line 820, in __getattr__
    module = self._get_module(self._class_to_module[name])
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/diffusers/utils/import_utils.py", line 832, in _get_module
    raise RuntimeError(
RuntimeError: Failed to import diffusers.pipelines.pipeline_utils because of the following error (look up to see its traceback):
Failed to import diffusers.models.autoencoders.autoencoder_kl because of the following error (look up to see its traceback):
Failed to import diffusers.loaders.unet because of the following error (look up to see its traceback):
Failed to import transformers.models.clip.modeling_clip because of the following error (look up to see its traceback):
/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/flash_attn_2_cuda.cpython-311-x86_64-linux-gnu.so: undefined symbol: _ZN2at4_ops5zeros4callEN3c108ArrayRefINS2_6SymIntEEENS2_8optionalINS2_10ScalarTypeEEENS6_INS2_6LayoutEEENS6_INS2_6DeviceEEENS6_IbEE
```

**Comment by tin2tin** [comment]
All of the dependencies will be/are installed in the path. And something is conflicting. The first ting to try is to click uninstall dependencies. When finished delete all folders with starting with ~ like ~orch. Then start Blender and Install Dependencies.

**Comment by jurassicjordan** [comment]
now when you say delete folders starting with ~, are you refering to folders within the site-packages folder? because I just uninstalled deps, but there doesnt seem to be folders there with a tilde

**Comment by tin2tin** [comment]
Yes, this folder: 
`/home/jordancruz/Tools/Blender-Versions/stable/blender-4.2.0-stable.a51f293548ad/4.2/python/lib/python3.11/site-packages/`

If you like, you can join the Pallaidium discord (invite on main page). There is a Linux user there, who's got Pallaidium up and running. Maybe he can help you, too. (I do not have Linux)

**Comment by Marksusu** [comment]
Here is the translated text:

---

I would like to share my experience of installing Pallaidium on an Archlinux system. Following the instructions on the official page, I performed the following steps:

- Install git (if your system already has git, you can skip this step)
- Download the add-on: https://github.com/tin2tin/text_to_video/archive/refs/heads/main.zip
- Run Blender
- In Blender, install the downloaded add-on

However, upon attempting to enable the Pallaidium plugin, I encountered an issue due to a dependency on the `aud` library, which is not available in my environment. 

That `aud` https://pypi.org/project/aud/ is:

> Support for Python 3.8, 3.7, & 3.6

My Archlinux system has Python version 3.12, so perhaps `aud` is not necessary.

After some troubleshooting with guidance from ChatGPT and inspired by the post at https://github.com/tin2tin/Pallaidium/issues/105 from @vuaieo, I found a solution by commenting out the problematic line in the code. 

Here's how:

- Navigate to the Pallaidium plugin directory:
   ```sh
   cd ~/.config/blender/4.2/extensions/user_default/pallaidium_generative_ai/
   ```

- Open the `__init__.py` file in a text editor and comment out the line importing `aud`:
   ```python
   ...
   import os
   # import aud  # Adding a "#" before it makes it a comment, and the program will skip this line.
   import re
   ...
   ```

- Save the file and restart Blender. Now you should be able to enable the Pallaidium plugin without any issues.

But hold on, there are two more crucial steps regarding dependencies that need to be addressed after creating a virtual environment. This step is essential "to avoid disrupting the system's existing standard environment (at least that’s what ChatGPT says)."

- In the Generative AI add-on preferences, click the "Uninstall Dependencies" button (to clear out any incompatible libraries).

No response? According to the official instructions, you should restart Blender at this point. 

Actually, a virtual environment should have been created before this step.

The so-called "virtual environment" refers to isolating another set of Python environments specifically for the Pallaidium plugin within a directory, separate from the system's standard environment.

Even if there was no response in the previous step, this next step is absolutely necessary, especially on an Arch Linux system where environmental requirements are very strict.

- In the Generative AI add-on preferences, click the "Install Dependencies" button. Without a virtual environment, you will encounter the following error:

   ```
   error: externally-managed-environment

   × This environment is externally managed
   ╰─> To install Python packages system-wide, try 'pacman -S python-xyz', where xyz is the package you are trying to install.

       If you wish to install a non-Arch-packaged Python package, create a virtual environment using 'python -m venv path/to/venv'. Then use path/to/venv/bin/python and path/to/venv/bin/pip.

   ...
   ```

In essence, it strongly suggests creating a virtual environment, which is an unavoidable prerequisite no matter how you look at it.

So, with the help of ChatGPT, I built a virtual environment (similar to what you need to do before installing ComfyUI), following these steps:

- **Enter the Blender Plugin Directory:**
   ```sh
   cd ~/.config/blender/4.2/extensions/user_default    # This is the directory where the "pallaidium_generative_ai" plugin resides ('~' represents 'home/your_user_name/'). Blender custom plugins are all located here, and we can also place the virtual environment directory for the Pallaidium plugin in this location.
   ```

- **Build the Virtual Environment:**
   ```sh
   python -m venv pall_venv    # This "pall_venv" is the name of the virtual environment built for the Pallaidium plugin; you can change it to any name you prefer.
   ```

- **Activate the Virtual Environment:**
   ```sh
   source ~/.config/blender/4.2/extensions/user_default/pall_venv/bin/activate
   ```

- **Start Blender from the Terminal:**
   ```sh
   blender
   ```

Next, you can proceed with the following step:

- In the Generative AI add-on preferences, click the "Install Dependencies" button.

In the virtual environment directory `~/.config/blender/4.2/extensions/user_default/pall_venv/`, the program will automatically install the necessary dependencies, and you can see the entire installation process in the terminal window.

After seeing a message similar to "installation completed," make sure to **activate the virtual environment from the terminal and then start Blender from the terminal**. This ensures that the Pallaidium plugin runs smoothly in an environment that satisfies all dependencies.

- **Activate the Virtual Environment:**
  ```sh
  source ~/.config/blender/4.2/extensions/user_default/pall_venv/bin/activate
  ```

- **Start Blender from the Terminal:**
  ```sh
  blender
  ```

Now, you can open the add-on UI in the Sequencer > Sidebar > Generative AI within the Video Editing workspace.

**Just a reminder:**

Every time you want to use Pallaidium in Blender, make sure to **activate the virtual environment from the terminal and then start Blender from the terminal**. This will ensure that the Pallaidium plugin runs smoothly in an environment that meets all dependencies.

you can combine the two commands into one line using `&&`, which ensures that Blender will only start if the virtual environment is successfully activated. Here's how it looks:

```sh
source ~/.config/blender/4.2/extensions/user_default/pall_venv/bin/activate && blender
```

This way, you can activate the virtual environment and start Blender in a single command.

---
> Hi there! I’m new to Arch Linux and only understand simple English. Plus, I barely know anything about Python programming. The steps above were all taught to me by ChatGPT, and some of them might have been done correctly just by chance. If you have any questions or run into issues, I won't be much help—feel free to reach out to AI for assistance! Looking forward to sharing experiences with you. 😊

**Comment by tin2tin** [comment]
@Marksusu @jurassicjordan @vuaieo 
A lot of work has gone into getting Pallaidium to work on Mac (however I still do not have macOS nor Linux), but maybe some of you could check how this affects running Pallaidium on Linux? There is also a Linux dedicated thread on the Pallaidium Discord. 

For quickly checking what is working on Linux and what is not working. This add-on can be used in combination with an installed version of Pallaidium. If you want to contribute, then share the reports added to the text editor, so I can add them to the Wiki pages. Ex. for MacOS: https://github.com/tin2tin/Pallaidium/wiki/macOS-Compatibility-Status 
Pallaidium Module Checker - for batch checking/downloading of supported genAI models in Pallaidium : 
https://github.com/tin2tin/pallaidium_module_checker

**Comment by tin2tin** [comment]
@limnetik Oh, that is good news. If you have time and hdd space could you, for Linux and Mac, run the https://github.com/tin2tin/pallaidium_module_checker 
and then grab the output in the Blender Text Editor, so I can add them to the status pages in the wiki?

The Old MacOS status:
https://github.com/tin2tin/Pallaidium/wiki/macOS-Compatibility-Status

MacOS runs a completely different lib, and I do not think I've implemented Kontext to that, but if a newer Diffusers is working, kontext will properly work on Linux and Win too.

**Comment by metaphysician** [comment]
hey there - updating the information as it applies to Arch Linux (CachyOS). the biggest issue to mention that will cause headaches is having Python 3.14 because the dependencies don't support it. 3.13 appears to be fine and in fact the beta version of Blender 5.2 portable includes it. i tried a number of workarounds and it's no go if any part of the system uses 3.14. there's a few other issues, but i'm trying to find the most straightforward way to install with a minimum of reconfiguration. i will update this post once i have a more definitive solution.

**Comment by tin2tin** [comment]
Thanks for giving it a shot and sharing. 

Yes, Blender 5.2 is locked at py version 3.13, and comes with it inside the python folder. It shouldn't be using any other version on the computer, but ex. pip is pesky looking for libs everywhere you do not want it to, if it doesn't find them in the side-packages folder inside the blender python folder.

**Comment by metaphysician** [comment]
okay i've gone back and forth on python 3.13 and there's basically no way to use that requirements.txt with python 3.13 either it seems. i've done it several times and there are all kinds of contradicting support issues resulting in errors when building.  many modules seem to want 3.11 and apparently up until 5.2 python 3.11 was preferred. at this point i'm giving up. i have it using only the version in site packages but for the moment 5.2 beta is a total no-go and no portable package can be downloaded using 3.11.

**Comment by tin2tin** [comment]
@metaphysician Let me understand this correctly, when you hit Install Dependencies the dependencies are not compatible with Python 3.13 on Linux? What if you remove all version numbers in requirements.txt, replace the file(restart Blender) and see what happens, how far it'll get, after you hit: Install Dependencies. 

If there are errors, then first round is just to uncomment the faulty libs (insert # before the pylibs name in the file), and try again.

NB. There is a difference between warnings(harmless), and errors(break the installation). Pip has a system writing a lot of info about presumed incompatibilities, but they are just harmless warnings - so only uncomment when the installation breaks because of an error. 

And then we'll end up with a list of troublemakers, and we can see how necessary they are, if it can be solved by being exact on version numbers etc. Send me a log or the requirements file with the uncommented troublemakers. 

I removed the version numbers here, so replace this with the similar named file in the Pallaidium folder: [requirements.txt](https://github.com/user-attachments/files/28663013/requirements.txt)

Blender 5.2 Beta for Linux: https://builder.blender.org/download/daily/

**Comment by tin2tin** [comment]
@metaphysician @limnetik I asked Claude for help, and now Pallaidium on Linux loads a dedicated requirements_linux.txt file. And here are the instructions on how to test it/track down the problems:

Linux (Arch / CachyOS / Ubuntu / etc.) — Blender 5.2 Installation Guide

1. Use the Blender 5.2 portable build

Download the Linux portable from https://builder.blender.org/download/daily/ and extract it. Do not use a system-installed Blender or a Flatpak/Snap — those won't use the correct bundled Python. Blender 5.2 ships Python 3.13 internally and the installer targets that.

2. Install the extension

In Blender: Edit → Preferences → Extensions → Install from Disk → select the Pallaidium .zip.

3. Click Install Dependencies

Open the Pallaidium panel in the Sequencer sidebar → go to the Dependencies section and click Install Dependencies. On Linux the installer automatically uses requirements_linux.txt (no pinned versions) and allows source builds where needed — no manual file editing is required for a first attempt.

Watch progress in Window → Toggle System Console.

4. If a package fails

The System Console will print a message like:


```
PALLAIDIUM pip FAILED — Base batch 3/12: mediapipe, onnxruntime
→ To skip, add '#' before each name in requirements_linux.txt and click Install Dependencies again.
Open the file requirements_linux.txt inside the Pallaidium extension folder (usually ~/.config/blender/5.2/extensions/user_default/pallaidium_generative_ai/) and comment out the failing package names with #:

# mediapipe
# onnxruntime
```
Then click Install Dependencies again. Repeat until the install completes.

5. Restart Blender

Once the install reports success, restart Blender and open the Pallaidium panel. 

Then send me the requirements_linux.txt file.

Known packages that may need commenting out on Python 3.13:

- mediapipe — no 3.13 wheel yet; affects pose/hand detection only
- onnxruntime — may need a nightly build; affects some audio/ONNX models
- jax / jaxlib — optional; only needed for JAX-based models

The core image/video generation pipeline works without these.

**Comment by metaphysician** [comment]
thanks for the information - i hadn't thought of removing the version numbers. i understand that warnings aren't as important as errors. the ones i seemed to remember having issues with were protobuf and wandb and laion. i think i got it to complete one time using my ComfyUI virtual env since that had Python 2.11, but it failed with generation on not finding torch c (or the version of torch it wanted anyway). there were lots of ups and downs. but the most recent was downloading the nightly 5.2 with 3.13 included and that didn't work with the requirements.txt that was included in the plugin. i tried getting Claude to edit the script to use packages for python 3.13 but that has not worked so far, even with leaving a few of the packages out.

i'll try the module list requirements file without version numbers tomorrow and see how far i get.

**Comment by tin2tin** [comment]
A huge thanks to jahjedi, who invested a lot of time on a complete walkthrough of how to get Pallaidium running on Linux. I've fed this into Claude and updated Pallaidium with the changes. So, fingers crossed. I hope it is working now...?

For reference, I've attached the walkthrough. 

[pallaidium_linux_installation_guide_en.md](https://github.com/user-attachments/files/28678169/pallaidium_linux_installation_guide_en.md)

**Comment by shiyue1250** [comment]
Hi! 👋 Linux user here — big thanks to @tin2tin and everyone in this thread for the groundwork.

I got Pallaidium generating reliably on Linux (Blender 5.2, NVIDIA), and since dependency install seems to be the main pain point, here's the exact recipe that worked for me:

**Verified core stack** (into Blender's bundled Python):
`torch 2.8 + cu12.8` · `diffusers 0.38` · `sdnq 0.2` · `transformers 4.57` · `opencv`
LTX-2.3 via `OzzyGT/LTX-2.3-Distilled-*` (sdnq int8), loaded with `enable_model_cpu_offload()` — a full `.to("cuda")` OOMs ~30 GB, offload keeps the peak ~10–15 GB.

**The non-obvious gotcha:** HuggingFace's Xet/CAS transport is blocked on some networks, so `snapshot_download` stalls on the large weight files. Forcing the real `huggingface.co` endpoint through an HTTP proxy fixes it (mirror endpoints only serve metadata).

I packaged all of this — a one-command Linux installer (replacing the in-addon button), a proxy-aware weight downloader, plus an optional **remote backend** (edit in Blender on a laptop, generate on a headless GPU box) — into a Linux-first fork, **Cineloom** (GPL-3.0, credits Pallaidium upstream): https://github.com/shiyue1250/cineloom

Happy to upstream any of the Linux install bits if useful. Thanks again for the great addon! 🙏
